docs/src/handle.rst

```diff
@@ -24,12 +24,23 @@ Data types
 .. c:type:: void (*uv_alloc_cb)(uv_handle_t* handle, size_t suggested_size, uv_buf_t* buf)
 
     Type definition for callback passed to :c:func:`uv_read_start` and
-    :c:func:`uv_udp_recv_start`. The user must fill the supplied :c:type:`uv_buf_t`
-    structure with whatever size, as long as it's > 0. A suggested size (65536 at the moment)
-    is provided, but it doesn't need to be honored. Setting the buffer's length to 0
-    will trigger a ``UV_ENOBUFS`` error in the :c:type:`uv_udp_recv_cb` or
+    :c:func:`uv_udp_recv_start`. The user must allocate memory and fill the supplied
+    :c:type:`uv_buf_t` structure. If NULL is assigned as the buffer's base or 0 as its length,
+    a ``UV_ENOBUFS`` error will be triggered in the :c:type:`uv_udp_recv_cb` or
     :c:type:`uv_read_cb` callback.
 
+    A suggested size (65536 at the moment in most cases) is provided, but it's just an indication,
+    the user is free to allocate the ammount of memory they decide.
```

Comment:
Thanks for chiming in! I'll add some more text. FWIW, we are removing this in v2, since it doesn't add much, but causes trouble.